### PR TITLE
Create SuperUser Class

### PR DESCRIPTION
**Pull request by JamesShadowfyre**
This class adds a SuperUser to the program, which gains premium features. 
Addresses issue #166 , waiting for issue #196 to be complete. 
PR is here to confirm pipeline issues addressed ahead of merge.

### CODE REVIEW

**Review by Birdbh — CHANGES_REQUESTED** [review]
This does not pass the testing pipeline. Upon closer inspection the short error summary is:
==================================================================== short test summary info ==================================================================== 
ERROR tests/test_superUser.py::test_logging_in - sqlite3.OperationalError: database is locked
ERROR tests/test_user.py::test_register - sqlite3.OperationalError: database is locked

**Comment by JamesShadowfyre** [comment]
> This does not pass the testing pipeline. Upon closer inspection the short error summary is: ==================================================================== short test summary info ==================================================================== ERROR tests/test_superUser.py::test_logging_in - sqlite3.OperationalError: database is locked ERROR tests/test_user.py::test_register - sqlite3.OperationalError: database is locked

I am aware, that's why it had no reviewers assigned. It is an interesting case, as the tests pass individually. 
Its only when run at a group that certain tests fail. Wanted to see if on repo it passed or not

File: src/flaskr/login.py [hunk 1]
```diff
@@ -14,6 +15,23 @@
     
     def login(self, username, password):
         db = get_db()
+        try:
+            userid = db.execute("SELECT id from user where username = (?)", (username,)).fetchone()['id']
+            try:
+                if(db.execute("SELECT userid from PremiumUser where userid = (?)", (userid,)).fetchone()['userid'] == userid):
+                    hasher = sha256()
+                    hasher.update(bytes(password, 'utf8'))
```
[the remark this answers is not in the document]
Comment (JamesShadowfyre) [reply src/flaskr/login.py:23]: why didnt it catch this before?

**Review by Birdbh — APPROVED** [review]
Looks good, when you merge though there will be multiple conflicts with main because home.html, home.py, Prediction, and a number of tests relating to those have passed. There should be no conflict between what you have done and those files though just that they have changes since what you did. You can either rebase now or resolve the merge conflicts when you merge.

**Review by Jcp149 — APPROVED** [review]
Looks great, very cool that we are hashing the passwords now like the flask tutorial

**Pull request merged**